reactor-core/src/main/java/reactor/core/publisher/ParallelFlux.java

```diff
@@ -1268,4 +1284,37 @@ else if (s2 != null) {
 		return both;
 	}
 
+	private static class MultiSubscriber<T> implements CoreSubscriber<T> {
+
+		final CoreSubscriber<T>[] subscribers;
+
+		private MultiSubscriber(CoreSubscriber<T>[] subscribers) {
+			this.subscribers = subscribers;
+		}
+
+		public CoreSubscriber<T>[] getSubscribers() {
+			return subscribers;
+		}
+
+		@Override
+		public void onSubscribe(Subscription s) {
+			throw new IllegalStateException("Not implemented");
```

Comment:
Prefer `UnsupportedOperationException`... But the fact we have UOE makes me wonder about the relevance of having a `MultiSubscriber`...

Reply:
TBH I have no idea how to do it otherwise

Reply:
Or... Maybe I do.

What if instead of applying the hook here, we apply it in `ParallelSource`/`ParallelArraySource` instead, where the actual subscription is happening?

Reply:
I don't think it works, `ParallelSource` being at the top of the chain (at least the segment that is parallelized)... `onLastOperator` is targeting the bottom of the chain, the point where we stop handing off to another operator's `Subscriber` and switch to user-space `Subscriber`